examples/standalone_mode_demo.ipynb

```diff
@@ -0,0 +1,609 @@
+    "# Graphstorm Standalone Mode Demonstration\n",
+    "\n",
+    "In this notebook, we'll demonstrate how to use the standalone mode of Graphstorm. The standalone mode is primarily designed for model developers to prototype their data and model training/evaluation/inference pipelines using a single machine. \n",
+    "\n",
+    "---\n",
+    "\n",
+    "## Setup \n",
+    "\n",
+    "This notebook requires installing graphstorm using pip. Please find [more details on installation of graphstorm](https://graphstorm.readthedocs.io/en/latest/install/env-setup.html#setup-graphstorm-with-pip-packages)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# !pip install graphstorm"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from argparse import Namespace\n",
+    "import matplotlib.pyplot as plt\n",
+    "# import graphstorm related stuffs\n",
+    "import graphstorm as gs\n",
+    "from graphstorm.config import GSConfig\n",
+    "from graphstorm.trainer import GSgnnNodePredictionTrainer\n",
+    "from graphstorm.dataloading import GSgnnNodeTrainData, GSgnnNodeDataLoader, GSgnnNodeInferData\n",
+    "from graphstorm.inference import GSgnnNodePredictionInferrer\n",
+    "from graphstorm.eval import GSgnnAccEvaluator"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# init the graphstorm \"cluster\"\n",
+    "def init_standalone_environment():\n",
+    "    \"\"\"Initialize standalone environment for graphstorm\"\"\"\n",
+    "    env_vars = {\n",
+    "        'RANK': '0',\n",
+    "        'WORLD_SIZE': '1',\n",
+    "        'MASTER_ADDR': '127.0.0.1',\n",
+    "        'MASTER_PORT': '54321'\n",
+    "    }\n",
+    "    # set up required ENV_VAR for distributed context\n",
+    "    for key, val in env_vars.items():\n",
+    "        os.environ[key] = val\n",
+    "\n",
+    "    gs.initialize(ip_config=env_vars['MASTER_ADDR'], backend='gloo')\n",
+    "\n",
+    "init_standalone_environment()\n",
+    "device = gs.utils.setup_device(0)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Develop a GNN model for node-classification \n",
+    "\n",
+    "Next, we use a GNN model to go over the model development cycle of a node classification problem on the MovieLens dataset. \n",
+    "\n",
+    "\n",
+    "### 0. Load the constructed graph data\n",
+    "\n",
+    "We can use the following codes to set up the `GSConfig` object by providing the yaml config file of the training script and the json config file about the constructed graph dataset. `GSConfig` handles the parsing of all the config files normally provided through command line. Therefore, we can simulate this in the Notebook using `argparse.Namespace`. "
```

Comment:
let's remove GSConfig. when people develop custom models, they don't have to use GSConfig to set up the hyperparameters. GSConfig is used for the command line tool.

Reply:
One big challenge of removing GSConfig comes from those settings to be used in model generation. If we want to use gs.create_builtin_node_gnn_model(), we need to pass a GSConfig object as an argument. If we do not use .create_builtin_node_gnn_model() method, then we need to expose the details of each layer, which will make this demo much complex.

Reply:
when i designed the API, i didn't intend to expose `create_builtin_node_gnn_model`. i wrote that as a helper function.
if we expose GSConfig, users need to learn the entire GSConfig. i wonder if this is necessary.

Reply:
Agree, I added a new comment, asking to use details in the `create_builtin_node_gnn_model` to build a model. We didn't expose the `create_builtin_node_gnn_model` API.